src/System.Private.CoreLib/src/System/StubHelpers.cs

```diff
@@ -1453,42 +1453,82 @@ internal struct NativeVariant
 #endif
     }  // struct NativeVariant
 
+    internal abstract class CleanupWorkListElement
+    {
+        private CleanupWorkListElement m_Next;
+        protected abstract void DestroyCore();
+
+        public void Destroy()
+        {
+            DestroyCore();
+            CleanupWorkListElement next = m_Next;
+            while (next != null)
+            {
+                next.DestroyCore();
+                next = next.m_Next;
+            }
+        }
+        
+        public static void AddToCleanupList(ref CleanupWorkListElement list, CleanupWorkListElement newElement)
```

Comment:
Are we doing to have a threading issue here? Is this state only accessed on a single thread?

Reply:
> Is this state only accessed on a single thread?

Yes.